main.py

```diff
@@ -0,0 +1,291 @@
+# TODO: Flush the cache using a webhook from Airtable when records are added or updated on various tables
+# TODO: Check the membership of the mentors internal channel when linking for a mentorship request
+
+# Start an asynchronous Slack Bolt application
+app = AsyncApp(
+    token=os.environ.get("SLACK_BOT_TOKEN"),
+    signing_secret=os.environ.get("SLACK_SIGNING_SECRET"),
+)
+
+# Currently, the app create functionality is not released in the Slack SDK for Python
+#  see https://github.com/slackapi/python-slack-sdk/issues/1119
+# try:
+#     bot_name = os.getenv("BOT_NAME")
+#     bot_username = os.getenv("BOT_USERNAME")
+#     ngrok_url = os.getenv("NGROK_URL")
+#     app_config_token = os.getenv("SLACK_APP_CONFIGURATION_TOKEN")
+#     if bot_name is None or bot_username is None or ngrok_url is None or app_config_token is None:
+#         raise "Please ensure you have all required environment variables set..."
+#
+#     synchronous_app = App(
+#         token=os.environ.get("SLACK_BOT_TOKEN"),
+#         signing_secret=os.environ.get("SLACK_SIGNING_SECRET"),
+#     )
+#     with open('bot_manifest.yml', 'r') as yaml_manifest:
+#         yaml_obj = yaml.safe_load(yaml_manifest)
+#         json_manifest = json.dumps(yaml_obj)
+#         synchronous_app.client.apps_manifest_create(manifest=json_manifest)
+#
+# except Exception as generic_error:
+#     sys.exit(1)
+
+# Define the application handler for the async Slack Bolt application - this adapter is specific to FastAPI
+app_handler = AsyncSlackRequestHandler(app)
+
+# Define the API
+api = FastAPI()
+
+# Initialize an AsyncIOScheduler object to schedule tasks
+Scheduler = AsyncIOScheduler({"apscheduler.timezone": "UTC"})
+Trigger = IntervalTrigger(seconds=30)
+
+
+# Start up our job scheduler on FastAPI startup and schedule jobs as needed
+@api.on_event("startup")
+async def startup_event() -> None:
+    Scheduler.start()
+    # job = Scheduler.add_job(schedule_messages, trigger=Trigger)
+    # logging.debug(f"Scheduled {job.name} with job_id: {job.id}")
+
+
+# On shutdown, shutdown the scheduler service first
+@api.on_event("shutdown")
+async def shutdown_event():
+    await Scheduler.shutdown()
+
+
+@api.post("/pybot/api/v1/slack/invite")
+async def invite_new_user(
+    email: str = Body(
+        ..., example="Test@test.com", description="Email address of the user to invite"
+    )
+) -> None:
+    await app.client.admin_users_invite(
+        team_id=SlackTeam.slack_id,
+        channel_ids=f"{SlackTeam.general_channel.id}",
+        email=email,
+    )
+
+
+# The base URI for Slack to communicate with our application - this URI is used for events, commands, and any other interaction
+@api.post("/slack/events")
+async def base_endpoint(req: Request) -> Response:
+    return await app_handler.handle(req)
+
+
+@app.command("/mentor_request")
```

Comment:
I've been exploring a way to divide these out into modules, but it's actually harder than you think. It requires a kind of wonky way to pass around the Slack request.